sdk/src/main/java/com/mapbox/maps/ResourceOptionsManager.kt

```diff
@@ -24,20 +24,51 @@ class ResourceOptionsManager(
    * Static variables and methods.
    */
   companion object {
-    private lateinit var default: ResourceOptionsManager
+    internal var default: ResourceOptionsManager? = null
+    internal fun getTokenResId(context: Context): Int = context.resources.getIdentifier(
+      MAPBOX_ACCESS_TOKEN_RESOURCE_NAME,
+      "string",
+      context.packageName
+    )
+
+    /**
+     * Convenience function to remove the default instance. Calling `getDefault` again will re-create the default instance.
+     */
+    internal fun destroyDefault() {
+      default = null
+    }
 
     /**
      * The default shared instance with default resource options.
+     *
+     * @param context the application context
+     * @defaultToken the default token will be set to the default [ResourceOptionsManager] instance.
+     * If set to null, will reuse the previously set one or search for an access token in the application resources for the first time of init.
+     * It will throw [MapboxConfigurationException] if there's no token found.
      */
     @Synchronized
-    fun getDefault(context: Context): ResourceOptionsManager {
-      if (!this::default.isInitialized) {
-        default = ResourceOptionsManager(
-          ResourceOptions.Builder().applyDefaultParams(context)
-            .build()
-        )
+    fun getDefault(context: Context, defaultToken: String? = null): ResourceOptionsManager {
+      // Apply the user setting token as the default token.
+      defaultToken?.let { token ->
+        val builder = ResourceOptions.Builder().applyDefaultParams(context).accessToken(token)
+        return ResourceOptionsManager(builder.build()).also { default = it }
       }
```

Comment:
There's a potential bug, if the default is not null, this will overwrite previously set default, right?

Reply:
we could add a test case for it.

Reply:
Yes, this is desigend as we described in the doc `the default token will be set to the default [ResourceOptionsManager] instance`
We have a test for it `getDefaultWithUserTokenTest`

Reply:
If user has changed other settings, for example cacheSize, in side the ResourceOptionsManager.default previously, this will reset their previous custom setting to mapbox default.

Reply:
Will reuse the previous config